be/src/Project/Web/Buyer/Buyer.Web/Areas/Products/Services/Products/ProductsService.cs

```diff
@@ -114,9 +114,27 @@ public async Task<PagedResults<IEnumerable<CatalogItemViewModel>>> GetProductsAs
             };
         }
 
-        public async Task<IEnumerable<string>> GetProductSuggestionsAsync(string searchTerm, int size, string language, string token)
+        public async Task<IEnumerable<ProductSuggestionViewModel>> GetProductSuggestionsAsync(string searchTerm, int size, string language, string token, string searchArea)
         {
-            return await this.productsRepository.GetProductSuggestionsAsync(searchTerm, size, language, token);
+            var products = await _productsRepository.GetProductsAsync(
+                null,
+                null,
+                null,
+                language,
+                searchTerm,
+                true,
+                PaginationConstants.DefaultPageIndex,
+                ProductConstants.ProductsCatalogPaginationPageSize,
+                token,
+                nameof(Product.Name));
+
+            return await Task.WhenAll(products?.Data?.Select(async x => new ProductSuggestionViewModel
```

Comment:
**suggestion:** I don't think Task.WhenAll makes sense here

Reply:
Hi Dawid, Task.WhenAll is required here because I use GetProductAttributesAsync method inside the Select. We tried figure this out with @SebaKas21 but this solution was the best.

If you have better solution for this case, please let me know.